addons/binding/org.openhab.binding.edimax/src/main/java/org/openhab/binding/edimax/handler/Edimax2101Handler.java

```diff
@@ -0,0 +1,117 @@
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.edimax.EdimaxBindingConstants;
+import org.openhab.binding.edimax.internal.commands.GetCurrent;
+import org.openhab.binding.edimax.internal.commands.GetPower;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Edimax2101Handler} is responsible for handling commands for the
+ * Edimax 2101, which are sent to one of the channels.
+ *
+ * @author Falk Harnisch - Initial contribution
+ */
+public class Edimax2101Handler extends EdimaxHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Edimax2101Handler.class);
+
+    private ScheduledFuture<?> pollingJob;
+
+    /**
+     * Constructor the only calls the super constructor.
+     *
+     * @param thing The thing
+     */
+    public Edimax2101Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        try {
+            if (channelUID.getId().equals(EdimaxBindingConstants.CURRENT)) {
+                if (command instanceof RefreshType) {
+                    final DecimalType current = new DecimalType(getCurrent());
+                    LOGGER.debug("Current: {}", current);
+                    updateState(channelUID, current);
+                }
+            }
+            if (channelUID.getId().equals(EdimaxBindingConstants.POWER)) {
+                if (command instanceof RefreshType) {
+                    final DecimalType power = new DecimalType(getPower());
+                    LOGGER.debug("Power: {}", power);
+                    updateState(channelUID, power);
+                }
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        final Runnable runnable = () -> refreshValues();
+        pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, 30, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        pollingJob.cancel(true);
+    }
+
+    /**
+     * Sends command to the channel to refresh current and power.
+     */
+    private void refreshValues() {
+        handleCommand(getThing().getChannel(EdimaxBindingConstants.CURRENT).getUID(), RefreshType.REFRESH);
```

Comment:
In my intention the `handleCommand `will only be called by the UI. But how are my values are updated if I want to use the current values in a rule?